e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,286 @@
+	require.ErrorContains(t, err, "the chain is paused")
+
+	// relay MsgRecvPacket & MsgAcknowledgement
+	require.NoError(t, r.Flush(ctx, eRep, ibcPathName, nobleToGaiaChannelID))
+
+	// uusdc IBC denom on gaia
+	srcDenomTrace := transfertypes.ParseDenomTrace(transfertypes.GetPrefixedDenom("transfer", gaiaToNobleChannelID, denomMetadataUsdc.Base))
+	dstIbcDenom := srcDenomTrace.IBCDenom()
+
+	gaiaWalletBal, err := gaia.GetBalance(ctx, gaiaWallet.FormattedAddress(), dstIbcDenom)
+	require.NoError(t, err)
+	require.True(t, gaiaWalletBal.IsZero())
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: IBC send TF token from a blacklisted user
+	// EXPECTED: Request fails;
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nobleWallet)
+
+	_, err = noble.SendIBCTransfer(ctx, nobleToGaiaChannelID, nobleWallet.KeyName(), transfer, ibc.TransferOptions{})
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not send tokens", nobleWallet.FormattedAddress()))
+
+	require.NoError(t, r.Flush(ctx, eRep, ibcPathName, nobleToGaiaChannelID))
+
+	gaiaWalletBal, err = gaia.GetBalance(ctx, gaiaWallet.FormattedAddress(), dstIbcDenom)
+	require.NoError(t, err)
+	require.True(t, gaiaWalletBal.IsZero())
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nobleWallet)
+
+	// ACTION: IBC send out a TF token to a blacklisted user
+	// EXPECTED: Request fails;
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, gaiaWallet)
+
+	_, err = noble.SendIBCTransfer(ctx, nobleToGaiaChannelID, nobleWallet.KeyName(), transfer, ibc.TransferOptions{})
+	require.ErrorContains(t, err, fmt.Sprintf("an address (%s) is blacklisted and can not receive tokens", gaiaWallet.FormattedAddress()))
+
+	require.NoError(t, r.Flush(ctx, eRep, ibcPathName, nobleToGaiaChannelID))
+
+	gaiaWalletBal, err = gaia.GetBalance(ctx, gaiaWallet.FormattedAddress(), dstIbcDenom)
+	require.NoError(t, err)
+	require.True(t, gaiaWalletBal.IsZero())
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, gaiaWallet)
+
+	// ACTION: IBC send out a TF token to malformed address
+	// EXPECTED: Request fails;
+
+	transfer.Address = "malformed-address1234"
+
+	_, err = noble.SendIBCTransfer(ctx, nobleToGaiaChannelID, nobleWallet.KeyName(), transfer, ibc.TransferOptions{})
+	require.ErrorContains(t, err, "decoding bech32 failed")
+
+	// ACTION: Successfully IBC send out a TF token to an address on another chain
+	// EXPECTED: Success;
+
+	transfer.Address = gaiaWallet.FormattedAddress()
+
+	_, err = noble.SendIBCTransfer(ctx, nobleToGaiaChannelID, nobleWallet.KeyName(), transfer, ibc.TransferOptions{})
+	require.NoError(t, err)
+
+	require.NoError(t, r.Flush(ctx, eRep, ibcPathName, nobleToGaiaChannelID))
+
+	gaiaWalletBal, err = gaia.GetBalance(ctx, gaiaWallet.FormattedAddress(), dstIbcDenom)
+	require.NoError(t, err)
+	require.Equal(t, transfer.Amount, gaiaWalletBal)
+}
```

Comment:
Any value in checking the noble wallet balance in any of the above cases? Or it's sufficient to just check the gaia balance?

Reply:
I think it is unnecessary as this is more testing core IBC functionality. However, I am happy to add the checks if you prefer.

Reply:
Ok, I think this is sufficient for me. Thanks!